allennlp/layers/embeddings.py

```diff
@@ -0,0 +1,173 @@
+    initialized to the pre-trained embeddings.  The Embedding layer can either be trainable or
+    not.
+
+    We use the DataIndexer to map from the word strings in the embeddings file to the indices
+    that we need, and to know which words from the embeddings file we can safely ignore.  If we
+    come across a word in DataIndexer that does not show up with the embeddings file, we give
+    it a zero vector.
+
+    The embeddings file is assumed to be gzipped, formatted as [word] [dim 1] [dim 2] ...
+    """
+    words_to_keep = set(vocab.get_index_to_token_vocabulary(namespace).values())
+    vocab_size = vocab.get_vocab_size(namespace)
+    embeddings = {}
+    embedding_dim = None
+
+    # TODO(matt): make this a parameter
+    embedding_misses_filename = 'embedding_misses.txt'
+
+    # First we read the embeddings from the file, only keeping vectors for the words we need.
+    logger.info("Reading embeddings from file")
+    with gzip.open(embeddings_filename, 'rb') as embeddings_file:
+        for line in embeddings_file:
+            fields = line.decode('utf-8').strip().split(' ')
+            if embedding_dim is None:
+                embedding_dim = len(fields) - 1
+                assert embedding_dim > 1, "Found embedding size of 1; do you have a header?"
+            else:
+                if len(fields) - 1 != embedding_dim:
+                    # Sometimes there are funny unicode parsing problems that lead to different
+                    # fields lengths (e.g., a word with a unicode space character that splits
+                    # into more than one column).  We skip those lines.  Note that if you have
+                    # some kind of long header, this could result in all of your lines getting
+                    # skipped.  It's hard to check for that here; you just have to look in the
+                    # embedding_misses_file and at the model summary to make sure things look
+                    # like they are supposed to.
+                    continue
+            word = fields[0]
+            if word in words_to_keep:
+                vector = numpy.asarray(fields[1:], dtype='float32')
+                embeddings[word] = vector
+
+    # Now we initialize the weight matrix for an embedding layer, starting with random vectors,
+    # then filling in the word vectors we just read.
+    logger.info("Initializing pre-trained embedding layer")
+    if log_misses:
+        logger.info("Logging embedding misses to %s", embedding_misses_filename)
+        embedding_misses_file = codecs.open(embedding_misses_filename, 'w', 'utf-8')
+    embedding_matrix = torch.FloatTensor(vocab_size, embedding_dim).uniform_(-0.05, 0.05)
+
+    # Depending on whether the namespace has PAD and UNK tokens, we start at different indices,
+    # because you shouldn't have pretrained embeddings for PAD or UNK.
+    start_index = 0 if namespace.startswith("*") else 2
```

Comment:
You should be able to always start at 0.  The check for `word in embeddings` will fail, and everything will be ok, because we just continue in that case.  And the namespace check is wrong here, anyways.